subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -22,8 +25,8 @@
 
 #include "hal/nrf_gpio.h"
 
-#if DT_NODE_HAS_STATUS(DT_PHANDLE(DT_NODELABEL(radio), coex), okay)
-#define CX_NODE DT_PHANDLE(DT_NODELABEL(radio), coex)
+#if DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_coex))
```

Comment:
Also, I find it confusing that "pin forwarder" and actual coex implementation share the same DTS description. Pin forwarder should simply use description in DTS which pins are to be connected to Net Core, while coex needs to know details that given pins are connected to external PTA, which pin is in/out, active high/low, pin role, etc.

Reply:
Generally I agree with you. This should be designed better, app should have list of pins in dts to forward, net core should simply use them.  However we are not sure how to solve pin forwarding in both elegant and easy for end user way. Consider `-DSHIELD` , consider overlays (for end user convenience possibly having just one overlay file per feature like cx should be considered), consider adding next pins to forward as a result of using features like "cx", fem" and whatever we need and what does not know about each other.
The case is also that we need this just now, we can open later other PR when we find way for better pin forwarding.

Reply:
I agree that the information both cores need from the devicetree description of Coex node differs and that the application core could do with pin numbers only. However, the approach where only a single .dts file is needed felt simpler to me. For example, a single .dts file describing a Coex shield could be applied to both cores. IMO it makes sense for a single shield to have a single devicetree description.

I'm aware that the entire concept of pin forwarder handling the pin assignments is somewhat crooked and should probably be handled in an entirely different manner. I don't think that having separate DTS descriptions changes the situation much - the solution is still not very elegant, but requires more effort to be used with shields conveniently.

Reply:
Using separate DTS node for describing pin forwarding does not prevent us from using single DTS overlay for a shield. I was thinking about something like:

```
// DTS shield overlay describing a part of external radio chip

&pin_forwarder {
  coex_pins = <&gpio0 7 &gpio1 8 &gpio1 12>;
  irq_pin = <&gpio1 11>
};

/ {
  radio_coex: radio_coex_three_wire {
    status = "okay";
    compatible = "generic-radio-coex-three-wire";
    req-gpios =     <&gpio0 7 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
    pri-dir-gpios = <&gpio1 8 (GPIO_PULL_UP   | GPIO_ACTIVE_HIGH)>;
    grant-gpios =   <&gpio1 12 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
  };
};

&radio {
  coex = <&radio_coex>;
};
```

This way we explicitly state which pins are going to be forwarded by App Core, instead of implicitly passing this through `generic-radio-coex-three-wire` node which should not be parsed by App Core at all.

Reply:
> Net Core would ignore pin_forwarder node, while App Core ignores radio_coex and radio

I'm not aware of a possibility to ignore parts of an overlay. AFAIK, an attempt to apply the above overlay to the application core will result in a build error because of a non-existing node `radio` being configured. Probably `pin_forwarder` would yield a similar error when applying that overlay to the network core.

Reply:
Yeah, probably should be described a little differently, like:
```
/{
  pin_forwarder {
    ...
  };

  radio {
    ...
  };
};
```

instead of using references. But it does not change the essence of my proposal